### ISSUE TITLE
CardScan crashes after minimize then maximize app

### ISSUE THREAD

**Issue by SOFSPEEL** [getbouncer/cardscan-android#52]
- open cardscan 

- minimizing app

- maximizing app

- app crahes

Occurs on following phone:
![image](https://user-images.githubusercontent.com/1612646/69557312-38f14d00-0f5b-11ea-9a92-ab4ac3dd5a38.png)

```


Thread finished: <Thread Pool> #22
[ScanActivityImpl] Prediction (ms): 1296
[skia] onFlyCompress
[ViewRootImpl@795b1dd[ScanActivityImpl]] MSG_WINDOW_FOCUS_CHANGED 0
[SensorManager] unregisterListener ::  
[Appboy v3.3.0 .com.appboy.Appboy] Closed session with ID: de348fdc-5403-4547-85f5-674363dc3673
[InputTransport] Input channel destroyed: fd=121
[ScanActivityImpl] Prediction (ms): 1535
[Appboy v3.3.0 .bo.app.bn] Completed the openSession call. Starting or continuing session de348fdc-5403-4547-85f5-674363dc3673
[SensorManager] registerListener :: 1, LSM6DS3 Accelerometer, 200000, 0,  
[AndroidRuntime] Shutting down VM
[GAv4] syncDispatchLocalHits timed out: java.util.concurrent.TimeoutException
[GAv4] Successfully bound to service but never got onServiceConnected callback
[AppCenterCrashes] Unhandled Exception from source=AppDomain
[AppCenterCrashes] Java.Lang.NullPointerException: Attempt to invoke virtual method 'void android.hardware.Camera.setPreviewDisplay(android.view.SurfaceHolder)' on a null object reference
[AppCenterCrashes]   --- End of managed Java.Lang.NullPointerException stack trace ---
[AppCenterCrashes] java.lang.NullPointerException: Attempt to invoke virtual method 'void android.hardware.Camera.setPreviewDisplay(android.view.SurfaceHolder)' on a null object reference
[AppCenterCrashes] at com.getbouncer.cardscan.base.ScanBaseActivity$CameraPreview.surfaceCreated(ScanBaseActivity.java:557)
[AppCenterCrashes] at android.view.SurfaceView.updateWindow(SurfaceView.java:652)
[AppCenterCrashes] at android.view.SurfaceView.onWindowVisibilityChanged(SurfaceView.java:266)
[AppCenterCrashes] at android.view.View.dispatchWindowVisibilityChanged(View.java:11085)
[AppCenterCrashes] at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
[AppCenterCrashes] at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
[AppCenterCrashes] at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
[AppCenterCrashes] at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
[AppCenterCrashes] at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
[AppCenterCrashes] at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1786)
[AppCenterCrashes] at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1490)
[AppCenterCrashes] at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7027)
[AppCenterCrashes] at android.view.Choreographer$CallbackRecord.run(Choreographer.java:927)
[AppCenterCrashes] at android.view.Choreographer.doCallbacks(Choreographer.java:702)
[AppCenterCrashes] at android.view.Choreographer.doFrame(Choreographer.java:638)
[AppCenterCrashes] at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:913)
[AppCenterCrashes] at android.os.Handler.handleCallback(Handler.java:751)
[AppCenterCrashes] at android.os.Handler.dispatchMessage(Handler.java:95)
[AppCenterCrashes] at android.os.Looper.loop(Looper.java:154)
[AppCenterCrashes] at android.app.ActivityThread.main(ActivityThread.java:6688)
[AppCenterCrashes] at java.lang.reflect.Method.invoke(Native Method)
[AppCenterCrashes] at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
[AppCenterCrashes] at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
[Mono] Loading reference 5 of /storage/emulated/0/Android/data/org.cancer.fundraising/files/.__override__/Microsoft.AppCenter.Crashes.dll asmctx DEFAULT, looking for System.Core, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
[Mono] Assembly Ref addref Microsoft.AppCenter.Crashes[0x7f79ae2d00] -> System.Core[0x7f602a9000]: 18
2019-11-25 07:40:24 [TRACE] (MvvmCross.Logging.MvxLog) TRACKING_DEBUG:SendException: Exception type: Java.Lang.NullPointerException
Message       : Attempt to invoke virtual method 'void android.hardware.Camera.setPreviewDisplay(android.view.SurfaceHolder)' on a null object reference
Stacktrace:

  --- End of managed Java.Lang.NullPointerException stack trace ---
java.lang.NullPointerException: Attempt to invoke virtual method 'void android.hardware.Camera.setPreviewDisplay(android.view.SurfaceHolder)' on a null object reference
at com.getbouncer.cardscan.base.ScanBaseActivity$CameraPreview.surfaceCreated(ScanBaseActivity.java:557)
at android.view.SurfaceView.updateWindow(SurfaceView.java:652)
at android.view.SurfaceView.onWindowVisibilityChanged(SurfaceView.java:266)
at android.view.View.dispatchWindowVisibilityChanged(View.java:11085)
at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1786)
at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1490)
at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7027)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:927)
at android.view.Choreographer.doCallbacks(Choreographer.java:702)
at android.view.Choreographer.doFrame(Choreographer.java:638)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:913)
at android.os.Handler.handleCallback(Handler.java:751)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:154)
at android.app.ActivityThread.main(ActivityThread.java:6688)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)



[AndroidRuntime] FATAL EXCEPTION: main
[AndroidRuntime] Process: org.cancer.fundraising, PID: 7837
[AndroidRuntime] java.lang.NullPointerException: Attempt to invoke virtual method 'void android.hardware.Camera.setPreviewDisplay(android.view.SurfaceHolder)' on a null object reference
[AndroidRuntime] at com.getbouncer.cardscan.base.ScanBaseActivity$CameraPreview.surfaceCreated(ScanBaseActivity.java:557)
[AndroidRuntime] at android.view.SurfaceView.updateWindow(SurfaceView.java:652)
[AndroidRuntime] at android.view.SurfaceView.onWindowVisibilityChanged(SurfaceView.java:266)
[AndroidRuntime] at android.view.View.dispatchWindowVisibilityChanged(View.java:11085)
[AndroidRuntime] at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
[AndroidRuntime] at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
[AndroidRuntime] at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
[AndroidRuntime] at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
[AndroidRuntime] at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1290)
[AndroidRuntime] at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1786)
[AndroidRuntime] at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1490)
[AndroidRuntime] at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7027)
[AndroidRuntime] at android.view.Choreographer$CallbackRecord.run(Choreographer.java:927)
[AndroidRuntime] at android.view.Choreographer.doCallbacks(Choreographer.java:702)
[AndroidRuntime] at android.view.Choreographer.doFrame(Choreographer.java:638)
[AndroidRuntime] at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:913)
[AndroidRuntime] at android.os.Handler.handleCallback(Handler.java:751)
[AndroidRuntime] at android.os.Handler.dispatchMessage(Handler.java:95)
[AndroidRuntime] at android.os.Looper.loop(Looper.java:154)
[AndroidRuntime] at android.app.ActivityThread.main(ActivityThread.java:6688)
[AndroidRuntime] at java.lang.reflect.Method.invoke(Native Method)
[AndroidRuntime] at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
[AndroidRuntime] at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
Thread finished: <Thread Pool> #9

```

**Comment by awushensky** [comment]
I'm able to reproduce this. Fix is in progress.

**Comment by awushensky** [comment]
Fix is in SDK version `1.0.5111`. Can you please test and confirm?

**Comment by awushensky** [comment]
@SOFSPEEL 1.0.5111 is now available in bintray.
